components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/services/TemplateManagerService.java

```diff
@@ -868,7 +991,8 @@ public void undeployTemplate(String templateType, String uuid) throws TemplateMa
      * @throws TemplateManagerException
      */
     public void undeploySiddhiApp(String uuid) throws TemplateManagerException {
-        // todo: implement
+        SiddhiAppApiHelper siddhiAppApiHelper = new SiddhiAppApiHelper();
+        siddhiAppApiHelper.delete("localhost",uuid);
```

Comment:
Why we need a host ? and why its hardcoded ? if needed use constants

Reply:
Fixed it by ba97bd5e5ac2fa22efaa3d0a89e7024ac8587491